dlt/common/schema/utils.py

```diff
@@ -154,6 +154,37 @@ def has_default_column_prop_value(prop: str, value: Any) -> bool:
     return value in (None, False)
 
 
+def is_compound_prop(prop: str) -> bool:
+    """Checks if a column property is compound."""
+    if prop in ColumnPropInfos:
+        return ColumnPropInfos[prop].compound is True
+    # for any unknown property
+    return False
+
+
+def remove_compound_props(
+    columns: TTableSchemaColumns, compound_props: List[str]
+) -> TTableSchemaColumns:
+    """Removes compound properties from all columns in place."""
+    non_compound = [prop for prop in compound_props if not is_compound_prop(prop)]
+    if non_compound:
+        raise ValueError(
+            f"The following properties are not compound properties: {', '.join(non_compound)}"
+        )
+
+    for column in columns.values():
+        is_nullable = is_nullable_column(column)
+        for prop in compound_props:
+            if not is_nullable:
```

Comment:
if someone sets `nullable`: then it is applied after this check so s/he will always see this warning (you remove compound hints, then merge new hints...) my take is to remove it